plugins/atomicfu/atomicfu-compiler/testData/box/PropertyDeclarationTest.txt

```diff
@@ -0,0 +1,25 @@
+@kotlin.Metadata
+public final class PropertyDeclarationTest {
+    // source: 'PropertyDeclarationTest.kt'
+    private final static @org.jetbrains.annotations.NotNull field a$FU: java.util.concurrent.atomic.AtomicIntegerFieldUpdater
+    private volatile volatile @kotlin.jvm.Volatile field a: int
+    private final static @org.jetbrains.annotations.NotNull field head$FU: java.util.concurrent.atomic.AtomicReferenceFieldUpdater
+    private volatile volatile @kotlin.jvm.Volatile @org.jetbrains.annotations.Nullable field head: java.lang.Object
+    private final @org.jetbrains.annotations.NotNull field lateIntArr: java.util.concurrent.atomic.AtomicIntegerArray
+    private final @org.jetbrains.annotations.NotNull field lateRefArr: java.util.concurrent.atomic.AtomicReferenceArray
+    static method <clinit>(): void
+    public method <init>(): void
+    public final static @org.jetbrains.annotations.NotNull method getA$FU(): java.util.concurrent.atomic.AtomicIntegerFieldUpdater
```

Comment:
Correct me, if I wrong. The idea for getters is that we replace calls to original getters with calls to replaced ones, the same with `IrGetField`s:
```
a.getAtomic() -> a.getAtomic$FU()
a.atomic -> a.atomic$FU
```
If the assumption is correct, then we should inherit visibilities for both fields and getters. Thus, fields will be always private, while getters inherit visibility of the original getter.

However, in ArithmeticsTest fields are public, while here they are private, while getters are always public, which tells me, that fields inherit original getter's visibility.

Please, correct me if my assumptions are incorrect are we need to make getters public for some reason - may be for binary compatibility.

Reply:
Thank you! That inconsistency in access modifiers was really a bug. Now all fields (volatile and fields updaters) are private, and their accessors inherit visibility of the original accessors.